src/components/Books/Form.js

```diff
@@ -1,11 +1,73 @@
-import React from 'react';
-
-const Form = () => (
-  <form className="book-form">
-    <input type="text" placeholder="Title" />
-    <input type="text" placeholder="Author" />
-    <button type="button" className="add-btn">Add Book</button>
-  </form>
-);
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../../redux/books/books';
+
+const Form = () => {
+  const [book, setBook] = useState({
+    title: '',
+    author: '',
+    category: '',
+  });
+
+  const { title, author, category } = book;
+  const dispatch = useDispatch();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    const newBook = {
+      id: uuidv4(),
+      title,
+      author,
+      category,
+    };
+
+    dispatch(addBook(newBook));
+    setBook({
+      title: '',
+      author: '',
+      category: '',
+    });
+  };
+
+  const handleChange = (e) => {
+    setBook({
+      ...book,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className="book-form">
+      <input
+        type="text"
+        placeholder="Title"
+        name="title"
+        value={title}
+        onChange={handleChange}
+        required
+      />
+      <input
+        type="text"
+        placeholder="Author"
+        name="author"
+        value={author}
+        onChange={handleChange}
+        required
+      />
+      <input
+        type="text"
+        placeholder="Category"
+        name="category"
+        value={category}
+        onChange={handleChange}
+        required
+      />
```

Comment:
- [ ] (Optional) It would be nice to have a `select` form control instead of a text field for the categories. That way, a user will choose from the available options you have created instead of inputting his own. This will make the categories' input more organized. 👍